src/constants/tokens.ts

```diff
@@ -359,14 +358,21 @@ export function isPolygon(chainId: number): chainId is ChainId.POLYGON | ChainId
   return chainId === ChainId.POLYGON_MUMBAI || chainId === ChainId.POLYGON
 }
 
-function getPolygonNativeCurrency(chainId: number) {
-  switch (chainId) {
-    case ChainId.POLYGON:
-      return MATIC_POLYGON
-    case ChainId.POLYGON_MUMBAI:
-      return MATIC_POLYGON_MUMBAI
-    default:
-      throw new Error('Not polygon')
+class PolygonNativeCurrency extends NativeCurrency {
+  equals(other: Currency): boolean {
+    return other.isNative && other.chainId === this.chainId
+  }
+
+  get wrapped(): Token {
+    if (!isPolygon(this.chainId)) throw new Error('Not Polygon')
+    const wrapped = WRAPPED_NATIVE_CURRENCY[this.chainId]
+    invariant(wrapped instanceof Token)
+    return wrapped
+  }
+
+  public constructor(chainId: number) {
+    if (!isPolygon(chainId)) throw new Error('Not Polygon')
```

Comment:
its interesting we mix `invariant` and throwing errors, both are the same pattern. just noting, oftentime invariant functions take a optional second part to put the message